### ISSUE TITLE
Cannot set cron expression with environment variables in prefect.yaml

### ISSUE THREAD

**Issue by JohnLemmonMedely** [PrefectHQ/prefect#11255]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

I can't seem to use an environment variable for the cron expression. This makes it difficult to parameterize my deployment for staging vs production. I ended up having to write a little `yq` script in our CI code to edit the yaml for staging, which is not ideal.

### Reproduction

```python3
prefect.yaml looks like this. Run `prefect deploy` to trigger the error.

deployments:
- name: build_ml_features
  schedule:
    cron: "{{ $DEPLOY_CRON }}"
  entrypoint: my_flow.py:start_flow
```

### Error

```python3
1 validation error for CronSchedule
cron
  Invalid cron string: "{{ $DEPLOY_CRON }}" (type=value_error)
```

### Versions

```Text
Version:             2.14.2
API version:         0.8.4
Python version:      3.10.13
Git commit:          3efa0219
Built:               Thu, Oct 26, 2023 5:12 PM
OS/Arch:             darwin/arm64
Profile:             dev
Server type:         server
```

**Comment by zzstoatzz** [comment]
hi @JohnLemmonMedely - thanks for the issue!

I believe this is more or less the same issue as #11189 - i.e. unable to template env vars into certain deployment fields.

we will look at this as soon as we can

**Comment by zhen0** [comment]
Closing as a duplicate to make it easier to track but we should check that any fix for #11189 addresses this also.
